src/main/scala/org/wikimedia/analytics/hdfstools/HdfsRsyncCLI.scala

```diff
@@ -76,44 +111,95 @@ object HdfsRsyncCLI {
+            .text("Remove verbosity from application logging (WARN)")
+
+        opt[Unit]("all-logs-info")
+            .optional()
+            .action((_, c) => c.copy(rootLogLevel = Level.INFO))
+            .text("Set root logger level to INFO (default: ERROR)")
+
+        opt[Unit]("all-logs-debug")
+            .optional()
+            .action((_, c) => c.copy(rootLogLevel = Level.DEBUG))
+            .text("Set root logger level to DEBUG (default: ERROR)")
+
+        opt[String]("log-file")
+            .optional()
+            .action((x, c) => c.copy(logFile = Some(x)))
+            .text("File to write logs instead of sending them to stdout")
+
+
 
         opt[Unit]('r', "recursive")
             .optional()
             .action((_, c) => c.copy(recurse = true))
             .text("Recurse into directories (default: false)")
 
-        opt[Unit]('p', "perms")
+        opt[Unit]('d', "dirs")
             .optional()
-            .action((_, c) => c.copy(preservePerms = true))
-            .text("Preserve permissions (default: false)")
+            .action((_, c) => c.copy(copyDirs = true))
+            .text("Copy directories without recursion (default: false)")
 
-        opt[Unit]('t', "times")
+        opt[Unit]('m', "prune-empty-dirs")
             .optional()
-            .action((_, c) => c.copy(preserveTimes = true))
-            .text("Preserve modification times (default: false)")
+            .action((_, c) => c.copy(pruneEmptyDirs = true))
+            .text("prune empty directory chains (default: false)")
 
-        opt[Long]("times-diff")
+
+
+        opt[Unit]("resolve-conflicts")
             .optional()
-            .action((x, c) => c.copy(acceptedTimesDiffMs = x))
-            .text("Milliseconds by which modificationTimes can differ and still be considered equal (default: 1000)")
+            .action((_, c) => c.copy(resolveConflicts = true))
+            .text("Resolve multi-source files/directories conflicts (default: false)\n" +
+                "\t\tNote: times and perms conflicts of merged-folders are resolved by default using the defined strategy.")
+
+        opt[Unit]("use-most-recent-time")
+            .optional()
+            .action((_, c) => c.copy(useMostRecentModifTimes = true))
+            .text("Use most recent modificationTime object instead of first listed src in conflicts (default: false)")
+
+
+
+        opt[Unit]("existing")
+            .optional()
+            .action((_, c) => c.copy(existing = false))
+            .text("Skip creating new files in dst (default: false)")
+
+        opt[Unit]("ignore-existing")
+            .optional()
+            .action((_, c) => c.copy(ignoreExisting = false))
+            .text("skip updating files that exist in dst (default: false)")
+
+        opt[Unit]('u', "update")
+            .optional()
+            .action((_, c) => c.copy(update = false))
+            .text("Skip files that are newer in dst (default: false)")
```

Comment:
--update == skip?

Reply:
Actually no: update = skip-newer - Meaning that if a file exists both in src and dst, and they are different, the update happens only if src is newer than dst.

Reply:
Ah, can you clarify that in the doc text?

Reply:
tried :)

Reply:
Looks great thank you!